src/meltano/core/plugin/settings_service.py

```diff
@@ -73,6 +73,17 @@ def __init__(
                     )
                     for var, value in self.project.active_environment.env.items()
                 }
+
+                # expand state_id_suffix
+                self.project.active_environment.state_id_suffix = expand_env_vars(
+                    self.project.active_environment.state_id_suffix,
+                    {
+                        **self.env_override,
+                        **self.project.dotenv_env,
+                    },
+                    raise_if_missing=strict_env_var_mode,
+                )
```

Comment:
Implementation related to #6770.

Reply:
@kgpayne - You and @cjohnhanson are our experts on config and env var parsing as of now. 😄 Can you take a look at the above and advise?

Reply:
Thanks for this @ReubenFrankel 🙌 I'm not surprised that env var expansion was a hurdle - we have been talking for a while about centralising env var handling onto its own service (#6253). Your approach is inline with the current state of env var expansion 👍 I was surprised that the `self.project.dotenv_env` was necessary, as I'd have expected this to be filled in already at the `project_settings_service` layer. I see it comes in stead from `get_with_metadata` when accessing 'normal' settings, so is needed here. Will put more thoughts along that line in its own comment...